packages/pipeline-editor/src/PipelineEditorWidget.tsx

```diff
@@ -561,6 +561,71 @@ export class PipelineEditor extends React.Component<
     });
   }
 
+  async handleLoadPipeline(): Promise<void> {
+    this.widgetContext.ready.then(() => {
+      let pipelineJson = this.widgetContext.model.toJSON();
+      const pipelineVersion: number =
+        +Utils.getPipelineAppdataField(pipelineJson, 'version') || 0;
+      if (pipelineVersion !== PIPELINE_CURRENT_VERSION) {
+        // pipeline version and current version are divergent
+        if (pipelineVersion > PIPELINE_CURRENT_VERSION) {
+          // in this case, pipeline was last edited in a "more recent release" and
+          // the user should update his version of Elyra to consume the pipeline
+          showDialog({
+            title: 'Load pipeline failed!',
+            body: (
+              <p>
+                This pipeline corresponds to a more recent version of Elyra{' '}
+                <br />
+                and cannot be used until Elyra has been upgraded.
+              </p>
+            ),
+            buttons: [Dialog.okButton()]
+          });
+          if (this.app.shell.currentWidget) {
+            this.app.shell.currentWidget.close();
+          }
+          return;
+        } else {
+          // in this case, pipeline was last edited in a "old" version of Elyra and
+          // it needs to be updated/migrated.
+          showDialog({
+            title: 'Migrate pipeline?',
+            body: (
+              <p>
+                This pipeline corresponds to an older version of Elyra and needs
+                to be migrated.
+                <br />
+                Although the pipeline can be further edited and/or submitted
+                after its update,
+                <br />
+                the migration will not be completed until the pipeline has been
+                saved within the editor.
+                <br />
+                <br />
+                Proceed with migration?
+              </p>
+            ),
+            buttons: [Dialog.cancelButton(), Dialog.okButton()]
+          }).then(result => {
+            if (result.button.accept) {
+              // proceed with migration
+              pipelineJson = PipelineService.convertPipeline(pipelineJson);
```

Comment:
Should we pass the pipeline version to the convert method and avoid having to look it up again?
```suggestion
              pipelineJson = PipelineService.convertPipeline(pipelineJson, pipelineVersion);
```

Reply:
I would prefer not to do this now, as the consumer not necessarily needs to know how the fields are handled. Having said that, we probably need a "pipeline utility class" to hide a lot of these details from the consumers.

Reply:
I'm confused by what you mean by the "consumer" here.  This code is already extracting the version info from the pipeline to determine that conversion is necessary in the first place.

Are you suggesting that pipelines might be "migrated" from another location and call `convertPipeline()` unconditionally?